### ISSUE TITLE
Unnecessary string allocations when logging

### ISSUE THREAD

**Issue by prokopst** [launchdarkly/android-client#60]
I noticed that the client does unnecessary string allocations on many places when logging. For example:
```
Timber.d("jsonVariation: returning variation: " + result + " flagKey: " + flagKey + " user key: " + userManager.getCurrentUser().getKeyAsString());
```
should be changed into something like this to avoid that:
```
Timber.d("jsonVariation: returning variation: %s flagKey: %s user key: %s", result, flagKey, userManager.getCurrentUser().getKeyAsString());
```

Timber has actually lint rules for that:
https://github.com/JakeWharton/timber#lint

**Comment by arun251** [comment]
Thanks @prokopst . We will make the suggested improvement.

**Comment by gwhelanLD** [comment]
Thanks @prokopst, this should now be resolved in the new V2.6.0 release.
